Source/HTTP/WinHttp/winhttp_connection.cpp

```diff
@@ -1045,13 +1030,37 @@ void WinHttpConnection::callback_status_data_available(
     // Read new data into buffer
     if (newBytesAvailable > 0)
     {
-        pRequestContext->m_responseBuffer.resize(newBytesAvailable);
+        // Stage 1 performance optimization: For requests with known Content-Length,
+        // use maxReceiveBufferSize instead of WinHttpQueryDataAvailable result
+        DWORD bytesToRead = newBytesAvailable;
+        
+        // Check if we have a known content length and a custom buffer size
+        if (pRequestContext->m_responseBodySize > 0)
+        {
+            size_t maxReceiveBufferSize = 0;
+            HRESULT hr = HCHttpCallRequestGetMaxReceiveBufferSize(pRequestContext->m_call, &maxReceiveBufferSize);
+            if (SUCCEEDED(hr) && maxReceiveBufferSize > 0)
+            {
+                // Use the larger of the two, but cap at remaining bytes to read
+                size_t remainingBytes = pRequestContext->m_responseBodyRemainingToRead;
+                bytesToRead = static_cast<DWORD>(std::min({
+                    static_cast<size_t>(maxReceiveBufferSize),
+                    remainingBytes,
+                    static_cast<size_t>(MAXDWORD)
+                }));
+                
+                HC_TRACE_INFORMATION(HTTPCLIENT, "WinHttpConnection [ID %llu] Using maxReceiveBufferSize=%zu for known content-length, reading %d bytes", 
```

Comment:
I don't really have a strong opinion but is this really an INFORMATION level trace? this seems really verbose?